File: backend/src/main/java/com/example/demo/service/TaskService.java
package com.example.demo.service;

import com.example.demo.dto.TaskDto;
import com.example.demo.entity.Task;
import com.example.demo.vo.TaskVo;

import java.util.List;

public interface TaskService {
    TaskVo createTask(TaskDto taskDto);
    TaskVo getTaskById(Long id);
    List<TaskVo> getAllTasks();
    TaskVo updateTask(Long id, TaskDto dto);
    void deleteTask(Long id);
}

File: frontend/src/services/TaskService.js
import api from './api';

class TaskService {
    async getAllTasks() {
        const response = await api.get('/listAll');
        return response.data;
    }

    async createTask(taskData) {
        const response = await api.post('/createTask', {
            ...taskData,
            completed: false
        });
        return response.data;
    }

    async updateTask(id, taskData) {
        const response = await api.put(`/updateTask/${id}`, taskData);
        return response.data;
    }

    async deleteTask(id) {
        const response = await api.delete(`/${id}`);
        return response.data;
    }

    async markAsCompleted(task) {
        return this.updateTask(task.id, {
            ...task,
            completed: true
        });
    }
}

export default new TaskService();
